src/main/java/hillel/spring/doctors/DoctorsRestController.java

```diff
@@ -56,41 +50,63 @@ public Doctor findDoctorByID(@PathVariable Integer id) {
         return service.findAll(letter, specialization);
     }
 
+    @GetMapping("/doctors/{doctorId}/schedule/{date}")
+    public Appointment findSchedule(@PathVariable Integer doctorId,
+                                    @PathVariable
+                                    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate date) {
+        return service.findOrCreateSchedule(doctorId, date);
+    }
+
 
+    //delete methods
     @DeleteMapping("doctors/{id}")
     @ResponseStatus(HttpStatus.NOT_FOUND)
 
     public ResponseEntity deleteDoctor(@PathVariable Integer id) {
-        service.findById(id).orElseThrow(DoctorIsNotFoundException::new);
+        service.findById(id).orElseThrow(() -> new DoctorIsNotFoundException(id));
         service.delete(id);
         return new ResponseEntity<>(HttpStatus.NO_CONTENT);
     }
 
+    //put methods
     @PutMapping("/doctors/{id}")
     public ResponseEntity<?> updateDoctor(@RequestBody DoctorInputDto doctorInputDto, @PathVariable Integer id) {
         Doctor doctor = dtoConverter.toModel(doctorInputDto, id);
         service.update(doctor);
         return ResponseEntity.noContent().build();
     }
 
+    //post methods
     @PostMapping("/doctors/{doctorId}/schedule/{date}/{hour}")
+    @Retryable(StaleObjectStateException.class)
```

Comment:
It is better to use this annotation on a service layer. (I now that in my example it was on a controller, but it is conceptually incorrect).